src/util/validate-url.ts

```diff
@@ -0,0 +1,38 @@
+interface UrlValidationResult {
+	isValidUrl: boolean,
+	reason?: string
```

Comment:
I feels the error code should be defined here instead in the route handler post.

Reply:
Sorry, not sure what do you mean by error code? ... It is either valid or not, and if not valid there could be a reason why it is not...

Reply:
Never-mind, didn't wrap my head around that `reason` is here, while `errorCode` is elsewhere. Probably should co-locate together. But it is alright.

Reply:
And reason probably is a more UI stuff, maybe better to sit inside html (e.g. for i18n purpose).

But nevermind, all good.